performance-tests/src/main/java/io/kroxylicious/benchmarks/InvokerDispatchBenchmark.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.benchmarks;
+
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.kafka.common.message.ApiVersionsRequestData;
+import org.apache.kafka.common.message.FetchRequestData;
+import org.apache.kafka.common.message.ProduceRequestData;
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.message.ResponseHeaderData;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ApiMessage;
+import org.apache.kafka.common.utils.ByteBufferOutputStream;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+import io.kroxylicious.filters.FourInterfaceFilter0;
+import io.kroxylicious.filters.FourInterfaceFilter1;
+import io.kroxylicious.filters.FourInterfaceFilter2;
+import io.kroxylicious.filters.FourInterfaceFilter3;
+import io.kroxylicious.proxy.filter.ArrayFilterInvoker;
+import io.kroxylicious.proxy.filter.FilterInvoker;
+import io.kroxylicious.proxy.filter.FilterInvokers;
+import io.kroxylicious.proxy.filter.KrpcFilter;
+import io.kroxylicious.proxy.filter.KrpcFilterContext;
+import io.kroxylicious.proxy.filter.SpecificFilterInvoker;
+
+// try hard to make shouldHandleXYZ to observe different receivers concrete types, saving unrolling to bias a specific call-site to a specific concrete type
+@Fork(value = 2, jvmArgsAppend = "-XX:LoopUnrollLimit=1")
+@Warmup(iterations = 10, time = 1)
+@Measurement(iterations = 5, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+public class InvokerDispatchBenchmark {
+
+    public enum Invoker {
+        array {
+            @Override
+            FilterInvoker invokerWith(KrpcFilter filter) {
+                return new ArrayFilterInvoker(filter);
+            }
+        },
+        specific {
+            @Override
+            FilterInvoker invokerWith(KrpcFilter filter) {
+                return new SpecificFilterInvoker(filter);
+            }
+        },
+        switching {
+            @Override
+            FilterInvoker invokerWith(KrpcFilter filter) {
+                return FilterInvokers.arrayInvoker(filter);
+            }
+        };
+
+        abstract FilterInvoker invokerWith(KrpcFilter filter);
+    }
+
+    @org.openjdk.jmh.annotations.State(Scope.Benchmark)
```

Comment:
I think we can import this?

Reply:
Yeah, done in 192acbdac013b16aa487d6022c3c02ed6fde9ccd